src/python/pants/engine/internals/build_files.py

```diff
@@ -163,6 +166,53 @@ async def ensure_address_family(request: OptionalAddressFamily) -> AddressFamily
     return request.ensure()
 
 
+@dataclass(frozen=True)
+class BUILDFileEnvironmentVariablesRequest:
+    file_content: FileContent
+
+
+class BUILDFileEnvVarExtractor(ast.NodeVisitor):
+    def __init__(self):
+        super().__init__()
+        self.env_vars: set[str] = set()
+
+    @classmethod
+    def get_env_vars(cls, file_content: FileContent) -> Sequence[str]:
+        obj = cls()
+        obj.visit(ast.parse(file_content.content, file_content.path))
+        return tuple(obj.env_vars)
+
+    def visit_Call(self, node: ast.Call):
+        is_env = isinstance(node.func, ast.Name) and node.func.id == "env"
+        for arg in node.args:
+            if is_env and isinstance(arg, ast.Constant):
+                self.env_vars.add(arg.value)
+                return
+            else:
+                is_env = False
+            self.visit(arg)
+        for kwarg in node.keywords:
+            self.visit(kwarg)
+
+
+@rule
+async def extract_env_vars(
+    request: BUILDFileEnvironmentVariablesRequest,
+    session_values: SessionValues,
+) -> EnvironmentVars:
+    # For BUILD file env vars, we only ever consult the local systems env. TODO: indicate lookups
+    # for bootstrap targets, so we can consult env vars in a normal fashion for regular targets.
```

Comment:
Ah, guess I need to resolve this, as I guess it is the root for the CI test failure.